File: 0042_Trapping Rain Water/TrappingRainWaterTwoPointers.java
class Solution {
    public int trap(int[] height) {
        int ans = 0, n = height.length;
        int l = 0, r = n - 1;
        int lMax = height[0], rMax = height[n-1];
        
        while(l < r) {
            if(lMax < rMax) {
                ans += lMax - height[l];
                lMax = Math.max(lMax, height[++l]);
            }else{
                ans += rMax - height[r];
                rMax = Math.max(rMax, height[--r]);
            }
        }
        
        return ans;
    }
}

File: 0042_Trapping Rain Water/TrappingRainWaterTwoPointers.cs
public class Solution {
    public int Trap(int[] height) {
        var ans = 0;
        var l = 0;
        var r = height.Length - 1;
        var l_max = height[l];
        var r_max = height[r];
        
        while(l < r)
        {
            if(l_max <= r_max)
            {
                ans += l_max - height[l];
                l_max = Math.Max(l_max, height[++l]);
            }
            else
            {
                ans += r_max - height[r];
                r_max = Math.Max(r_max, height[--r]);
            }
        }
        
        return ans;
    }
}